snippets/formulae/route-pathfinder.py

```diff
@@ -0,0 +1,37 @@
+import heapq
+
+# This is a simplified implementation of the A* algorithm, and only serves as a reference.
+def find_path(origin_system, destination_system):
+    open_list = []
+    closed_list = {}
+    heapq.heappush(open_list, (0, None, origin_system))
+
+    while len(open_list) > 0:
+        current_cost, current_parent, current_system = heapq.heappop(open_list)
+        if current_system in closed_list:
+            # This system was already seen with a lower cost.
```

Comment:
It was seen, but how do we know it had a lower cost? If we find a lower cost path to a system we already found a path before, how will we update our list to it?